### PR TITLE
DAOS-1939 mgmt: drpc module and handler registration

### PR DESCRIPTION
**Pull request by tanabarr**
Register mgmt drpc module to communicate with control plane and
create a simple handler to verify functionality.

### CODE REVIEW

File: src/mgmt/srv.c [hunk 1]
```diff
@@ -35,6 +35,9 @@
 #include "srv_internal.h"
 #include <signal.h>
 
+/** Fully qualified path to daos_server socket */
+char *ds_mgmt_server_socket_path;
```
Comment (kjacque) [inline src/mgmt/srv.c]: Is the mgmt module going to be making dRPC client calls into the daos_server? If so, we should probably start thinking of how to centralize the client connections if we have multiple modules independently trying to talk as clients on the same socket. If we're just listening for calls from daos_server into daos_io_server, though, this isn't needed -- the IO server has a listener thread that already knows its listening socket.

File: src/mgmt/srv_internal.h [hunk 2]
```diff
@@ -52,4 +53,25 @@ void ds_mgmt_hdlr_tgt_create(crt_rpc_t *rpc_req);
...
+ * These numeric designations are used in dRPC communications in the Drpc__Call
+ * structure.
+ */
+
+/**
+ *  Module: Mgmt Server
+ *
+ *  The server module that deals with client mgmt requests.
+ */
+#define DRPC_MODULE_MGMT_SERVER				2
```
Comment (kjacque) [inline src/mgmt/srv_internal.h]: Shouldn't redefine the module ID here - use the one in drpc_modules.h instead

File: src/mgmt/srv.c (hunk 1 above)
Comment (tanabarr) [reply src/mgmt/srv.c]: nothing springs to mind as most information will probably be requested from daos__server. maybe events in the future?

Comment (kjacque) [reply src/mgmt/srv.c]: Might be best to remove in the short-term then.

**Comment by tanabarr** [comment]
duplicated and review comments addressed on #149

**Comment by brianjmurrell** [comment]
@tanabarr Please delete the branch if you no longer need it.

**Pull request closed without merging**